app/views/play/level/tome/SpellView.coffee

```diff
@@ -705,6 +718,42 @@ module.exports = class SpellView extends CocoView
     return if @destroyed
     Backbone.Mediator.publish 'tome:hide-problem-alert', {}
 
+  saveSpade: =>
+    return if @destroyed
+    spadeEvents = @spade.compile()
+    # Uncomment the below line for a debug panel to display inside the level
+    #@spade.debugPlay(spadeEvents)
+    condensedEvents = @spade.condense(spadeEvents)
+    return unless condensedEvents.length
+    compressedEvents = LZString.compressToUTF16(JSON.stringify(condensedEvents))
+
+    codeLog = new CodeLog({
+      sessionID: @options.session.id
+      level:
+        original: @options.level.get 'original'
+        majorVersion: (@options.level.get 'version').major
+      levelSlug: @options.level.get 'slug'
+      userID: @options.session.get 'creator'
+      log: compressedEvents
+    })
+    codeLog.save()
+
+    codeLog.on('save:success', (codelog) ->
+      codelogs = @options.session.get('codeLogs')
+      if codelogs? and codelogs.push?
+        codelogs.push codelog.id
+      else
+        codelogs = [codelog.id]
+      @options.session.set('codeLogs', codelogs)
+      @options.session.save()
+    , @)
```

Comment:
This would be better to do on the server side, as it would be more consistent, performant and succinct.

1. Create a middleware file for codelogs in `/server/middleware`
1. Copy the `rest` post middleware and make it specific to CodeLog.
1. Add logic to push the ID of the newly created `codeLog`'s id onto the session: `yield LevelSession.update({_id: mongoose.Types.ObjectId(req.body.sessionID)}, {$push: {codeLogs: codeLog._id}})`
1. Have `routes/index.coffee` use the custom middleware instead of the generic rest middleware for POST.